gpii/node_modules/flowManager/src/ReadyGetHandler.js

```diff
@@ -28,8 +28,8 @@ fluid.defaults("gpii.flowManager.cloudBased.ready.handler", {
 gpii.flowManager.cloudBased.ready.handler.reportError = function (that, prefsServerURL) {
     that.events.onError.fire({
         isError: true,
-        message: "Cannot connect to Preferences Server at " + prefsServerURL,
-        statusCode: 404
+        message: "Error connecting to Preferences Server at " + prefsServerURL,
```

Comment:
The preferences server is intended for internal use by CBFM and PPT etc. It won't be accessed by the public. I'm thinking for security reasons, it's probably better not to include prefs server URL in the response message, instead, before firing this event at line 29, use `fluid.log()` to log a detailed error, with the prefs server url, in the flow manager log file for developers and system admins.

This comment applies across the pull request for not exposing internal urls to the public, such as urls of prefs server and couchdb, but log them for debugging using `fluid.log()`.

Reply:
Okay, but that means we've had a security issue with the response error message -- I changed the status code only.

I will audit the code in this pull and make sure that couch and prefs urls are not exposed in responses.  What about flow manager urls?  Are they safe in the responses?  I believe they are, but I'll check.

Reply:
Yes, it was me who wrote this code that included the prefs server url in this response message. Sorry and thanks for fixing it.

The Cloud Based Flow manager URL should be OK to expose.

Reply:
@cindyli, I found another case that exposes the URL. When running the code in the [pull request for production configuration tests](https://github.com/GPII/universal/pull/718), the following error message is added to the payload when trying to update a snapset:

> `Cannot update:  GPII key "testUser1" is a snapset while executing HTTP PUT on url http://preferences:9081/preferences/%gpiiKey?merge=%merge`

I found the first part of that error message in [preferencesService.js](https://github.com/GPII/universal/blob/master/gpii/node_modules/preferencesServer/src/preferencesService.js#L291), namely this part: `Cannot update:  GPII key "testUser1" is a snapset`, but I can't find where the URL is added.

Any ideas?

Reply:
The second part is likely from [this line in kettle](https://github.com/fluid-project/kettle/blob/master/lib/dataSource-url.js#L257).

Reply:
Thanks!  I used grep on the universal code, but didn't think to look inside the top-level 'node_modules' folder.

Changing something in kettle is beyond the scope of this JIRA/PR, IMO.  Furthermore, the very next line in that kettle code removes the sensitive parts of the URL.  I think we are fine with exposing the URL in this case.

I've made the other two changes, and haven't found any other sensitive URL exposure.  What's left?